CONTRIBUTING_EN.md

```diff
@@ -156,33 +225,301 @@ Run all the tests with `make test` and make sure that they all passed.
+
+- Excessive technical details: Overly technical information that is irrelevant to the end user or does not provide context on the impact of the change should be avoided.
+Example: “Changed memory management in class X.”
+
+- Maintainer entries: Changes related only to the development or internal maintenance process, such as CI/CD tool configuration adjustments, are generally not relevant for the changelog.
+Example: “Updated GitHub Actions configuration.”
+
+- A bug introduced and fixed in the same release does not need a changelog entry.
+
+Avoid including this information to keep the changelog focused and useful for project users and contributors.
 
-### 11. Add changelog entries
+#### Writing good changelog entries
 
-[Add a changelog entry][keep-a-changelog] to CHANGELOG.md.
+A good changelog entry should be both descriptive and concise. It should explain the change to a reader with no prior context about the change. If it’s hard to be both concise and descriptive, lean toward being more descriptive.
 
-### 12. Create a GitHub PR
+- **Bad** : Go to the project order.
+- **Good**: Display the user’s starred projects at the top of the “Go to project” dropdown.
+
+The first example provides no context on where the change was made, why it was made, or how it benefits the user.
+
+- **Bad**: Copy (some text) to the clipboard.
+- **Good**: Update the tooltip of “Copy to clipboard” to indicate what is being copied.
+
+Again, the first example is too vague and provides no context.
+
+- **Bad**: Fix and improve CSS and HTML issues in the mini pipeline graph and builds dropdown.
+- **Good**: Fix tooltips and hover states in the mini pipeline graph and builds dropdown.
+
+The first example is too focused on implementation details. Users don’t care that we changed CSS and HTML, they care about the final result of those changes.
+
+- **Bad**: Remove null values in the Commit object array returned by find_commits_by_message_with_elastic.
+- **Good**: Fix 500 errors caused by Elasticsearch results referencing commits already collected by the garbage collector.
+
+The first example focuses on how we fixed something, not on what was fixed. The rewritten version clearly describes the final benefit to the user (fewer 500 errors) and when it happens (when searching for commits with Elasticsearch).
+
+Use your best judgment and try to put yourself in the position of someone reading the compiled changelog. Does this entry add value? Does it provide context on where and why the change was made?
+
+#### How to add an entry to the changelog
+
+The changelog is available in the [CHANGELOG.md][changelog] file.
+
+First, you need to identify the type of your change. Types of changes:
+
+- Added for new features. 
+- Changed for changes to existing features. 
+- Deprecated for features that will soon be removed. 
+- Fixed for any bug fixes. 
+- Removed for features that were removed. 
+- Security in case of vulnerabilities.
```

Comment:
```suggestion
- `Added` for new features. 
- `Changed` for changes to existing features. 
- `Deprecated` for features that will soon be removed. 
- `Fixed` for any bug fixes. 
- `Removed` for features that were removed. 
- `Security` in case of vulnerabilities.
```